internal/client.go

```diff
@@ -342,13 +397,13 @@ func (c *Client) OnRequestCompleted(rc RequestCompletionCallback) {
 
 // Do performs request passed
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) error {
-	// check if client exists or not
-	if c.client == nil {
+	// check if httpClient exists or not
+	if c.httpClient == nil {
 		return fmt.Errorf(c.ErrorMsg)
 	}
 
 	req = req.WithContext(ctx)
```

Comment:
That's interesting. You use request at the time API is invoked not when request is created. Should you pass context at all to functions which create request?

Reply:
No, removed them alotgether.